guidelines/commit_and_pr_text.md

```diff
@@ -0,0 +1,53 @@
+# Motivation
+Our changelogs (reviewed by users/customers) are autogenerated from PR titles and git commit text.  The better this information is, the more we help our customers.  This information also helps reviewers understand changes, patch managers understand risk/importance, and when hunting for regressions, helps us better understand what is/isn’t likely to be a cause.
```

Comment:
On the SDK team we recently went through this process of changelog. We used to have a single changelog file that would need to be edited for EACH PR. This caused a bunch of subsequent PRs to have merge conflicts as the file was updated.

We also thought about making the commit messages the source of truth. This proved to be quite annoying as it was a departure from our current workflow, meant much more diligence to the commit messages.

Our final solution was to create the changelog from a collection of fragments that were added to each PR. This allowed us to better describe the change for the customer/user in the changelog.

https://github.com/operator-framework/operator-sdk/pull/2761

Reply:
the changelog isn't a committed file in the repo, it's auto generated doc that's available w/ the release.

Reply:
I think that the kubernetes PR template https://github.com/kubernetes/kubernetes/edit/master/.github/PULL_REQUEST_TEMPLATE.md (a form of which CRI-O has adopted https://github.com/cri-o/cri-o/blob/master/.github/PULL_REQUEST_TEMPLATE.md) works well for generating changelog. It includes a `release-note` section that can be parsed to generate a changelog. It helps when a PR has multiple commits. The whole template itself is very useful to a reviewer for understanding the motivation for a PR.